### ISSUE TITLE
New pattern for a bucket hat

### ISSUE THREAD

**Issue by darigovresearch** [freesewing/freesewing#1431]
**What is it that you would like to see happen?**

Bucket hats are simple and versatile hat made of only cloth and may be a simple first hat for people who are looking to get started which can be adapted in various ways

https://en.wikipedia.org/wiki/Bucket_hat

**Are you a FreeSewing patron?**

 - [ ] Yes, I am :hugs:
 - [x] No, I am not :thinking:

**Additional context**

This will likely be our first pattern to see what the contribution process is like & how to generate parametric patterns. Anybody who is new to contributing or interested in helping is welcome to join. See this video if you need some guidance on how you may be able to help - https://www.youtube.com/watch?v=UWA4wyacY2A

Suggested name is Billy the bucket hat

**Comment by joostdecock** [comment]
This is very exciting. There might be people on our Discord server who'd love to somehow follow this effort or learn from it.
Shall I announce it there and point them to this issue, or how do you see this?

Also, I'd be very interested to hear about any pain-points or friction you encounter as a contributor so that we can strive to make the process as welcoming as possible.

Thank you for doing this.

**Comment by darigovresearch** [comment]
@joostdecock yes do feel free to use the announcements channel but we can also post in the discord if you set us up with the admin rights as well.

Happy to let you know about any pain points as a contributor or any other suggestions to make the process easier for people.

Also, we have some other ideas for patterns which other people will be welcome to participate or try out themselves which will be coming soon as other issues.

**Comment by joostdecock** [comment]
@darigovresearch if you like, you can announce it yourself in `#development-help` which is where the target audience will hang out anyway 👍

**Comment by joostdecock** [comment]
Perhaps cross-post to `#pattern-design` just to cover all bases

**Comment by darigovresearch** [comment]
Thanks for the guidance, sounds good & will do!

**Comment by bobgeorgethe3rd** [comment]
@darigovresearch 
https://bobgeorgethe3rd-billy-bucket-hat-sewing-pattern.netlify.app/#draft
My brain cracked the code as it were last night and drafted this. I am now realising that you may have been wanted to do this yourself so I thought I would check before putting out for testing as I will drop it if you do. Believe it or not this was not as simple as it seems, I had to crack out simultaneous equations to figure out a curve which I have not used in years XD.

**Comment by darigovresearch** [comment]
@bobgeorgethe3rd no worries at all, the more collaborators on a project the better it will be. Would love to see the code & simultaneous equations!

The demo looks good so far, the only thing that I can see is that when clicking on the "Test Billy" tab the top crown is not featured when selecting any of the parameters on the left hand side. Is there a reason for this?

**Comment by bobgeorgethe3rd** [comment]
Yea it uses the data-circle on point to draw the circle for quickness. It's not a path so that why doesn't show. I can change it to a path but it's just a lot more then doing a quick and dirty point data-circle. 

The simultaneous equations

hc = (r + crownLength)*angle
tc = r*angle

the r (radius) and angle are unknown. so you solve to find

angle = (hc - tc)/crownLength
r = tc/angle

Gives you the angle in rads and the radius so you can draft the arc of the crown side.

Right well I will see about putting this out for pattern testing in the Discord.

**Comment by darigovresearch** [comment]
Thanks for the equations & the explanation. From our perspective not having the top crown on the test feature is not necessary if it complicates the code to have it, not sure if there are any other contributors which have any opinions on that.

Not sure how feasible these are but potential additional nice to haves as optional extras in the pattern are
- snaps to snap the brim to the crown side for better visibility & ventilation - https://freesewing.dev/tutorials/pattern-design/completing-your-pattern/#adding-snippets
- small holes in the crown side to put chin laces to prevent hat from flying off in windy conditions

Having these autogeneratable so user doesn't need to think about where to punch holes and aligning them would be of great help but can be defaulted to none to make the code easier
